### DISCUSSION TITLE
Suggestion/Feature Request: Code Scanning for Groovy

### DISCUSSION THREAD

**Discussion by MPWTraining** [community/community#29960] [Code Security] [answered]
I'd like to suggest code scanning for Groovy as a feature for GitHub.  Though Groovy may not seem to hit the top charts for popularity, it is an enabler for Java platforms.

Currently it seems that CodeQL is only supported for C++, C#, Go, Java, javascript, Python and Ruby.

**Comment by aeisenberg** [comment, marked as answer]
Thank you for your suggestion. Groovy is not currently on our roadmap to support, but I will mention this with our product planning team.

**Reply by machad** [reply]
Groovy is used to write Jenkins pipelines. As sophisticated CI/CD pipelines become more popular, it becomes more valuable to have Groovy supported in CodeQL. I ask that this be raised with your product team again in this context.

**Comment by ChristopherMank** [comment]
Would love to see this as well! @aeisenberg - Any roadmap updates possible on this?

**Reply by aeisenberg** [reply]
Nothing on the roadmap right now.

**Reply by ChristopherMank** [reply]
Thanks for the reply

**Comment by vsrins1979** [comment]
All our Microservices are written in Groovy and currently we cant leverage Github advanced security tools for our Groovy projects and appreciate if you have a roadmap in 2025 supporting Groovy.
